src/TeamSpeak3QueryApi/FileTransferClient.cs

```diff
@@ -25,62 +24,47 @@ public async Task SendFile(byte[] data, int port, string key)
         {
             using var client = new TcpClient();
             await client.ConnectAsync(_host, port).ConfigureAwait(false);
-            var ns = client.GetStream();
+            using var ns = client.GetStream();
 
             // Send key
             var keyBytes = Encoding.ASCII.GetBytes(key);
             await ns.WriteAsync(keyBytes, 0, keyBytes.Length).ConfigureAwait(false);
-            ns.Flush();
+            await ns.FlushAsync().ConfigureAwait(false);
 
             // Send data
             await ns.WriteAsync(data, 0, data.Length).ConfigureAwait(false);
         }
 
-        public async Task SendFile(Stream dataStream, long size, int port, string key)
+        public async Task SendFile(Stream dataStream, int port, string key)
         {
             using var client = new TcpClient();
             await client.ConnectAsync(_host, port).ConfigureAwait(false);
-            var ns = client.GetStream();
+            using var ns = client.GetStream();
 
             // Send key
             var keyBytes = Encoding.ASCII.GetBytes(key);
-            ns.Write(keyBytes, 0, keyBytes.Length);
-            ns.Flush();
+            await ns.WriteAsync(keyBytes, 0, keyBytes.Length).ConfigureAwait(false);
+            await ns.FlushAsync().ConfigureAwait(false);
 
             // Send data
-            var bytesRemaining = size;
-            var buffer = new byte[4096];
-            do
-            {
-                // Read from source. Cast is safe because buffer.Length is Int32.
-                var bytesRead = await dataStream.ReadAsync(buffer, 0, (int) Math.Min(buffer.Length, bytesRemaining)).ConfigureAwait(false);
-
-                // Write into network stream
-                await ns.WriteAsync(buffer, 0, bytesRead).ConfigureAwait(false);
-
-                bytesRemaining -= bytesRead;
-            } while (bytesRemaining > 0);
+            await dataStream.CopyToAsync(ns).ConfigureAwait(false);
         }
 
-        public async Task<byte[]> ReceiveFile(int size, int port, string key)
+        public async Task<MemoryStream> ReceiveFile(int size, int port, string key)
```

Comment:
I'd perfer to return a more general type like `Stream` here, so we keep the ability to change the concrete implementation later.